src/viper/trans.ml

```diff
@@ -41,15 +63,16 @@ let rec unit (u : Mo_def.Syntax.comp_unit) : prog =
 and dec_fields (ctxt : ctxt) (ds : M.dec_field list) =
   match ds with
   | [] ->
-    (ctxt, [])
+    (ctxt, [], [])
   | d :: ds ->
-    let ctxt, mk_i = dec_field ctxt d in
-    let ctxt, mk_is = dec_fields ctxt ds in
-    (ctxt, mk_i::mk_is)
+    let ctxt, init, mk_i = dec_field ctxt d in
```

Comment:
`ctxt` gets shadowed below. Use `_`?

Reply:
I think @crusso used meant to use shadowing in this case, right?
As in, we first transform `ctxt` via `let ctxt, init, mk_i = dec_field ctxt d` and then via `let ctxt, inits, mk_is = dec_fields ctxt ds`.

Reply:
I am fine with it when it is intentional :-)